src/Microsoft.DotNet.Darc/DarcLib/GitHubClient.cs

```diff
@@ -39,16 +40,16 @@ public class GitHubClient : RemoteRepoBase, IRemoteGitRepo
     private static readonly Regex PullRequestUriPattern =
         new(@"^/repos/(?<owner>[^/]+)/(?<repo>[^/]+)/pulls/(?<id>\d+)$");
 
-    private readonly Lazy<IGitHubClient> _lazyClient;
     private readonly IRemoteTokenProvider _tokenProvider;
     private readonly ILogger _logger;
     private readonly JsonSerializerSettings _serializerSettings;
     private readonly string _userAgent = $"DarcLib-{DarcLibVersion}";
+    private IGitHubClient? _lazyClient = null;
```

Comment:
should we rename this to something like `_cachedGitHubClient` since it's no longer using the `Lazy<>` implementation? might be misleading like this

Reply:
But lazy still means you initialize it once, not that you use some `Lazy<T>` class